pyo3-macros-backend/src/module.rs

```diff
@@ -237,6 +237,7 @@ pub fn pymodule_module_impl(
                     )? {
                         set_module_attribute(&mut item_mod.attrs, &full_name);
                     }
+                    item_mod.attrs.push(parse_quote!(#[pyo3(submodule)]));
```

Comment:
I think this needs some additional tests.

If `submodule` is already present, will this error? Probably it will error for duplicate `submodule` setting already, the error message should ideally read "submodules of declarative modules don't need `submodule` attribute set".

We might want to allow for `submodule = false` or maybe `#[pymodule(importable)]` to allow turning this off. Maybe? (TBH I can't see a use case where users would want to nest declarative modules and then allow both to be importable as root modules too.)

Reply:
Yeah I cannot think of a valid use case for that scenario. If one _does_ want it, it can be accomplished by declrating two top-level modules and exporting one into the other.